src/materials/ConcreteExpansionMicrocrackingDamage.C

```diff
@@ -0,0 +1,167 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/*                       BlackBear                              */
+/*                                                              */
+/*           (c) 2017 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "ConcreteExpansionMicrocrackingDamage.h"
+#include "ElasticityTensorTools.h"
+
+registerMooseObject("BlackBearApp", ConcreteExpansionMicrocrackingDamage);
+
+template <>
+InputParameters
+validParams<ConcreteExpansionMicrocrackingDamage>()
+{
+  InputParameters params = validParams<ScalarDamageBase>();
+  params.addClassDescription("Scalar damage model based on extent of internal expansion");
+
+  params.addRequiredParam<MaterialPropertyName>(
+      "eigenstrain_name", "name of the eigenstrain driving the microcracking damage process");
+
+  params.addParam<bool>("use_isotropic_expansion",
```

Comment:
How about 'assume_isotropic_expansion'?

Also, 'decomposes the expansion alongside the principal direction' -> 'computes linear expansion based on the first principal eigenstrain'